server/src/main/java/io/crate/replication/logical/action/TransportDropSubscriptionAction.java

```diff
@@ -90,4 +108,26 @@ protected ClusterState execute(ClusterState currentState, DropSubscriptionReques
     protected ClusterBlockException checkBlock(DropSubscriptionRequest request, ClusterState state) {
         return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_WRITE);
     }
+
+    private Metadata.Builder removeSubscriptionSetting(Metadata.Builder mdBuilder, @Nonnull String subscriptionToDrop) {
+        Iterator<IndexService> indicesIterator = indicesService.iterator();
+        while (indicesIterator.hasNext()) {
+            IndexService indexService = indicesIterator.next();
+            var settings = indexService.getMetadata().getSettings();
+            var subscriptionName = REPLICATION_SUBSCRIPTION_NAME.get(settings); // Can be null.
```

Comment:
every `IndexService` can have only one subscription with the name given in `REPLICATION_SUBSCRIPTION_NAME`.
One subscription can have many publications with many tables each - and there are can be many IndexService instances satisfying  `subscriptionToDrop.equals(subscriptionName)` condition - so I'm not doing break in the loop.